### PR TITLE
fix: unable to resolve nodejs document loader in react native environment

### LINKED ISSUE #993: Unable to resolved https module in react-native
Unable to resolved `https` module from `/node_modules/@digitalcredentials/js onld/lib/documentLoaders/node.js `

I am trying to use PPV2 in react-native from branch `0.3.0-pre`, Once I installed the packages and trying to run the app I am getting below error.

Screenshot 2022-08-18 at 3 31 52 PM (1)

### PR DESCRIPTION
**Pull request by karimStekelenburg**
Fixes #993 by conditionally importing the XHR and NodeJS document loaders, based on the runtime environment.

### CODE REVIEW

File: packages/core/src/modules/vc/W3cCredentialService.ts [hunk 1]
```diff
@@ -304,10 +304,10 @@ export class W3cCredentialService {
 
       let loader
 
-      if (isNodeJS()) {
-        loader = documentLoaderNode.apply(jsonld, [])
-      } else if (isReactNative()) {
-        loader = documentLoaderXhr.apply(jsonld, [])
+      if (isReactNative()) {
```
[the remark this answers is not in the document]
Comment (karimStekelenburg) [reply packages/core/src/modules/vc/W3cCredentialService.ts:307]: Definitely not intending to go back to that and we should probably fix this by separate extensions in the future. This is just a quick fix for now ;)

File: packages/core/src/modules/vc/W3cCredentialService.ts [hunk 2]
```diff
@@ -302,15 +303,9 @@ export class W3cCredentialService {
...
-      let loader
-
-      if (isNodeJS()) {
-        loader = documentLoaderNode.apply(jsonld, [])
-      } else if (isReactNative()) {
-        loader = documentLoaderXhr.apply(jsonld, [])
-      } else {
-        throw new AriesFrameworkError('Unsupported environment')
-      }
+      // fetches the documentLoader from documentLoader.ts or documentLoader.native.ts depending on the platform at bundle time
```
Comment (TimoGlastra) [inline packages/core/src/modules/vc/W3cCredentialService.ts:306]: The isNodeJS and isReactNative methods are defined but never used. Can we remove those methods @karimStekelenburg ?

image

File: packages/core/src/utils/environment.ts [hunk 3]
```diff
@@ -5,5 +5,5 @@ export function isNodeJS() {
 export function isReactNative() {
```
Comment (TimoGlastra) [inline packages/core/src/utils/environment.ts:5]: Are these still used?

File: packages/core/src/modules/vc/W3cCredentialService.ts (hunk 2 above)
Comment (karimStekelenburg) [reply packages/core/src/modules/vc/W3cCredentialService.ts:306]: Nope, removed

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**